Remotion/ObjectBinding/Web/UI/Controls/BocListImplementation/Rendering/BocColumnRendererBase.cs

```diff
@@ -106,61 +106,76 @@ protected virtual bool HasContentAttribute
+      renderingContext.AddAttributeToRender(HtmlTextWriterAttribute.Class, cssClassTitleCell);
+      renderingContext.AddAttributeToRender(HtmlTextWriterAttribute2.Role, HtmlRoleAttributeValue.ColumnHeader);
       if (_renderingFeatures.EnableDiagnosticMetadata)
       {
         var columnItemID = renderingContext.ColumnDefinition.ItemID;
         if (!string.IsNullOrEmpty(columnItemID))
-          renderingContext.Writer.AddAttribute(DiagnosticMetadataAttributes.ItemID, columnItemID);
+          renderingContext.AddAttributeToRender(DiagnosticMetadataAttributes.ItemID, columnItemID);
 
         var columnTitle = renderingContext.ColumnDefinition.ColumnTitleDisplayValue;
-        HtmlUtility.ExtractPlainText(columnTitle).AddAttributeTo(renderingContext.Writer, DiagnosticMetadataAttributes.Content);
+        renderingContext.AddAttributeToRender(DiagnosticMetadataAttributes.Content, HtmlUtility.ExtractPlainText(columnTitle));
 
         var oneBasedCellIndex = renderingContext.VisibleColumnIndex + 1;
-        renderingContext.Writer.AddAttribute(DiagnosticMetadataAttributesForObjectBinding.BocListCellIndex, oneBasedCellIndex.ToString());
+        renderingContext.AddAttributeToRender(DiagnosticMetadataAttributesForObjectBinding.BocListCellIndex, oneBasedCellIndex.ToString());
 
-        renderingContext.Writer.AddAttribute(
+        renderingContext.AddAttributeToRender(
             DiagnosticMetadataAttributesForObjectBinding.BocListColumnHasContentAttribute,
             HasContentAttribute.ToString().ToLower());
 
-        renderingContext.Writer.AddAttribute(
+        renderingContext.AddAttributeToRender(
             DiagnosticMetadataAttributesForObjectBinding.BocListColumnIsRowHeader,
             arguments.IsRowHeader.ToString().ToLower());
       }
-      renderingContext.Writer.RenderBeginTag(HtmlTextWriterTag.Th);
-
-      RenderTitleCellMarkers(renderingContext);
-      RenderBeginTagTitleCellSortCommand(renderingContext);
-      RenderTitleCellText(renderingContext);
-      if (renderingContext.Control.IsClientSideSortingEnabled || renderingContext.Control.HasSortingKeys)
-        RenderTitleCellSortingButton(renderingContext, arguments.SortingDirection, arguments.OrderIndex);
-      RenderEndTagTitleCellSortCommand(renderingContext);
-
-      renderingContext.Writer.RenderEndTag();
     }
 
     void IBocColumnRenderer.RenderDataColumnDeclaration (BocColumnRenderingContext renderingContext, bool isTextXml)
     {
+      ArgumentUtility.CheckNotNull("renderingContext", renderingContext);
+
       RenderDataColumnDeclaration(new BocColumnRenderingContext<TBocColumnDefinition>(renderingContext), isTextXml);
     }
 
-    protected virtual void RenderDataColumnDeclaration (BocColumnRenderingContext<TBocColumnDefinition> renderingContext, bool isTextXml)
+    private void RenderDataColumnDeclaration (BocColumnRenderingContext<TBocColumnDefinition> renderingContext, bool isTextXml)
     {
```

Comment:
I am unsure if `RenderDataColumnDeclaration` should also be changed in this PR @MichaelKetting ?

Reply:
yes, it's mentioned somewhere during the first review by me. reasoning is that while it might not have been mentioned in the spec, this api is a sibling of the other two render-apis and should be covered by the same rules/styles. whenever we notice things like that after the task is started, it should be included, unless there's a good reason against it, such as it being significantly more work etc.